src/nvim/lua/executor.c

```diff
@@ -2217,3 +2218,46 @@ char *nlua_read_secure(const char *path)
 
   return buf;
 }
+
+void ex_trust(exarg_T *eap)
+{
+  char *mode;
+  char *path;
+  {
+    char *p = skiptowhite(eap->arg);
+    size_t len = (size_t)(p - eap->arg);
+    mode = xcalloc(len + 1, sizeof(char));
+    memcpy(mode, eap->arg, len);
+    path = skipwhite(p);
+  }
+
+  {
+    lua_State *const lstate = global_lstate;
+    lua_getglobal(lstate, "vim");
+    lua_getfield(lstate, -1, "secure");
+    lua_getfield(lstate, -1, "trust");
+    lua_pushstring(lstate, path);
+    lua_pushstring(lstate, mode);
+    if (nlua_pcall(lstate, 2, 2)) {
+      nlua_error(lstate, _("Error executing vim.secure.trust: %.*s"));
+      return;
+    }
+
+    size_t len = 0;
+    const char *contents = lua_tolstring(lstate, -1, &len);
+    if (contents != NULL) {
+      // Add one to include trailing null byte
+      char *errmsg = xcalloc(len + 1, sizeof(char));
+      memcpy(errmsg, contents, len + 1);
+      if (errmsg != NULL) {
+        semsg("Error: %s", errmsg);
```

Comment:
Rather than the generic `Error: `, let's define a new error number and message in `globals.h`:

```c
EXTERN char e_trustfile[] INIT(= N_("E5570: Cannot update trust file: %s"));
```

Used here like this:

```c
semsg(e_trustfile, errmsg);
```

And then update the docs for the `:trust` command to add a link to `|E5570|`.

Reply:
Where should I put the docs for `:trust`?

Reply:
Added the error message in 92d83872c23bfafd33760cb5ace870338edd0c64

Reply:
>Where should I put the docs for :trust?

Ultimately I don't think it matters *too* much since most people will just find it using the tag (i.e. using `:h :trust`). So I'll say just add a new section in `editing.txt`.

Reply:
Added docs in 597a586208219b7578f8b4bd2d38aa148af66e87